src/pretix/static/pretixcontrol/js/ui/editor.js

```diff
@@ -943,22 +948,40 @@ var editor = {
             editor.load(editor.history[editor.history.length - 1 - editor._history_pos]);
             editor._history_modification_in_progress = false;
             editor.dirty = true;
+            editor._update_save_button();
+        }
+    },
+
+    _update_save_button() {
+        if ($("#editor-save span").prop("disabled")) {
+            // Currently saving
+            return;
+        }
+        if (editor.dirty || !editor._ever_saved) {
+            $("#editor-save").removeClass("btn-success").addClass("btn-primary").find(".fa").attr("class", "fa fa-fw fa-save");
+        } else {
+            $("#editor-save").addClass("btn-success").removeClass("btn-primary").find(".fa").attr("class", "fa fa-fw fa-check");
```

Comment:
Should we update the headline as well? Note this code would update the headline only if the editor was not changed while saving and uses the current value of name … a proper way would be to pass `payload.name` to `editor._update_save_button()` and always set it instead of only setting it when editor is not dirty.
```suggestion
            $("#editor-save").addClass("btn-success").removeClass("btn-primary").find(".fa").attr("class", "fa fa-fw fa-check");
            $("h1 small").text($("#pdf-info-name").val())
```

Reply:
I thought about this, the problem is that the template does not just show the ``name`` in that ``<small>`` element, it shows ``title``, which is a property that plugins can override to do something like ``Badge-Layout {name}`` or ``Ticket-Layout {name}``, to show which type of layout this is. I don't think it's worth it replicating that logic (with it's extensibility) on the client side.